week2/index.js

```diff
@@ -1,65 +1,129 @@
+//Include File File System package.
 var fs = require('fs');
 
+//Get input commands, remove 'node index.js' form the command.
 var options = process.argv.slice(2);
 
-var command = options[0];
-
-switch (command) {
-  case 'help':
-  default:
-    showHelp();
-    break;
-  case 'list':
-    listTodos();
-    break;
+// find the value of the command, call help file in case command value not found.
+switch (options[0]) {
+	case 'help':
+	default:
+		showHelp();
+		break;
+	case 'list':
+		listTodos();
+		break;
+	case 'add':
+		addLine();
+		break;
+	case 'remove':
+		deleteLine();
+		break;
+	case 'reset':
+		resetFile();
+		break;
+	case 'update':
+		updateLine();
+		break;
 }
-
+//separate by new line file content, and filter it in array.
 function splitStringByNewline(string) {
-  return string.split('\n').filter(function(element) {
-    element = element.trim();
-    return element.length > 0;
-  });
+	return string.split('\n').filter(function (element) {
+		element = element.trim();
+		return element.length > 0;
+	});
 }
 
+//Call help file and display content.
 function showHelp() {
-  openFile('help.txt', function(error, data) {
-    if (error) {
-      return console.log('Error: the help file could not be displayed', error);
-    }
-    console.log(data);
-  });
+	var data = openFile('help.txt')
+	console.log(data);
 }
 
+//call todo file content and display its content.
 function listTodos() {
-  openFile('todo.txt', function(error, data) {
-    if (error) {
-      if (error.code === 'ENOENT') {
-        return console.log('Nothing to do! (or your dog ate your todo list)');
-      } else {
-        return console.log('Error: Something went wrong', error);
-      }
-    }
+	var data = openFile('todo.txt');
+	var todos = splitStringByNewline(data);
+	if (todos.length === 0) {
+		return console.log('Nothing to do!')
+	}
+	console.log('Your todo list looks like this');
+	todos.forEach(function (element, index) {
+		index = (index + 1).toString();
+		console.log(index, element);
+	});
+
+	if (todos.length > 5) {
+		console.log('You have too much to do!');
+	}
+
+}
+//Add new todo line, creat a file in case its not exist.
+function addLine() {
+	options.shift(); //remove first element from command
+	var allLine = options.join(' ') ; //wrap all command as one line
+	fs.appendFileSync('todo.txt', allLine + '\n', 'utf8'); // append the new line to list
+	listTodos(); // show the new list
+}
 
-    var todos = splitStringByNewline(data);
+//update new todo line, creat a file in case its not exist.
+function updateLine() {
+	if (options[1] <= 0) { // in case that the user enter worng number
+		console.log('Please enter right line number!!! ' + '\n');
+		showHelp();
+	} else {
+		var lineNumber = options[1] - 1; // get the line number
```

Comment:
All of this relies on `options` being a global object. But what if we want `options` to be read/evaluated inside a function, and thus not in global scope - how would your code need to change?